src/overlays/actors/ovl_En_Ma4/z_en_ma4.c

```diff
@@ -21,78 +94,1011 @@ const ActorInit En_Ma4_InitVars = {
+        || (this->poppedBalloonCounter == 10)) {
+        gSaveContext.unk_3DD0[4] = 6;
+        EnMa4_SetupHorsebackGameEnd(this, globalCtx);
+        D_80AC0258 = 0;
+    }
+}
+
+void EnMa4_SetupHorsebackGameEnd(EnMa4* this, GlobalContext* globalCtx) {
+    gSaveContext.weekEventReg[0x8] &= (u8)~0x01;
+    this->actionFunc = EnMa4_HorsebackGameEnd;
+    func_801A89A8(0x100000FF);
+    func_801A89A8(0x8041);
+}
+
+void EnMa4_HorsebackGameEnd(EnMa4* this, GlobalContext* globalCtx) {
+    static s32 sFrameCounter = 0;
+    Player* player = PLAYER;
+
+    if (player->stateFlags1 & 0x100000) {
+        globalCtx->actorCtx.unk268 = 1;
+        globalCtx->actorCtx.unk278 = 0x8000;
+    } else {
+        globalCtx->actorCtx.unk268 = 1;
+    }
+
+    if (sFrameCounter == 25) {
+        if (this->poppedBalloonCounter == 10) {
+            func_801518B0(globalCtx, 0x334F, &this->actor);
+            this->textId = 0x334F;
+        } else {
+            func_801518B0(globalCtx, 0x334B, &this->actor);
+            this->textId = 0x334B;
+        }
+    } else if (sFrameCounter == 50) {
+        globalCtx->actorCtx.unk268 = 0;
+        globalCtx->nextEntranceIndex = 0x6410;
+        gSaveContext.unk_3F4A = 0;
+        sFrameCounter = 0;
+        globalCtx->sceneLoadFlag = 0x14;
+
+        if (this->poppedBalloonCounter == 10) {
+            globalCtx->unk_1887F = 0x50;
+            gSaveContext.nextTransition = 3;
+        } else {
+            globalCtx->unk_1887F = 0x40;
+            gSaveContext.nextTransition = 2;
+        }
+
+        this->poppedBalloonCounter = 0;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/func_80ABDD2C.asm")
+    sFrameCounter++;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/func_80ABDD9C.asm")
+void EnMa4_SetupBeginEponasSongCs(EnMa4* this) {
+    this->actionFunc = EnMa4_BeginEponasSongCs;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/func_80ABDE60.asm")
+// Epona's Song cutscene is an ActorCutscene
+void EnMa4_BeginEponasSongCs(EnMa4* this, GlobalContext* globalCtx) {
+    s16 cutsceneIndex = this->actor.cutscene;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/EnMa4_Init.asm")
+    if (ActorCutscene_GetCanPlayNext(cutsceneIndex) != 0) {
+        ActorCutscene_Start(cutsceneIndex, &this->actor);
+        EnMa4_SetupEponasSongCs(this);
+    } else {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        }
+        ActorCutscene_SetIntentToPlay(cutsceneIndex);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/EnMa4_Destroy.asm")
+void EnMa4_SetupEponasSongCs(EnMa4* this) {
+    this->animTimer = 0;
+    this->actionFunc = EnMa4_EponasSongCs;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Ma4_0x80ABDCA0/func_80ABE1C4.asm")
+static u16 D_80AC0260 = 99;
```

Comment:
```suggestion
static u16 eponaSongCinemaLastAction = 99;
```

Reply:
The `Cinema` part could be shortened.
What about this?
```suggestion
static u16 sEponasSongCsLastAction = 99;
```
Besides that, I like the name.